packages/reaction-collections/server/publications/products.js

```diff
@@ -7,12 +7,31 @@
 Meteor.publish("Products", function (productScrollLimit, productFilters) {
   check(productScrollLimit, Match.OneOf(null, undefined, Number));
```

Comment:
We could simplify it like this:
```js
const filters = new SimpleSchema({
  "shops": {type: [String], optional: true},
  "tag": {type: String, optional: true},
  "query": {type: String, optional: true},
  "visibility": {type: Boolean, optional: true},
  "details": {type: Object, optional: true},
  "details.key": {type: String},
  "details.value": {type: String},
  "price": {type: Object, optional: true},
  "price.min": {type: Number},
  "price.max": {type: Number},
  "weight": {type: Object, optional: true},
  "weight.min": {type: Number},
  "weight.max": {type: Number}
})

Meteor.publish("Products", function (productScrollLimit, productFilters) {
  new SimpleSchema({
    productScrollLimit: { type: Number, optional: true },
    productFilters: { type: filters, optional: true }
  }).validator()
...
```
I think it should work not only in `validationMethod`. see [docs](https://github.com/aldeed/meteor-simple-schema#validating-and-throwing-validationerrors).  We don't need try/catch